spec/upload_to_s3_spec.rb

```diff
@@ -145,39 +145,141 @@ def stub_s3_response_for_file(key, exists: true)
+              bucket: test_bucket,
+              key: 'existing-key-2',
+              file: file_path,
+              skip_if_exists: false
+            )
+          end.to raise_error(FastlaneCore::Interface::FastlaneError, "File already exists in S3 bucket #{test_bucket} at #{expected_key}")
+        end
+      end
+
+      it 'fails when if_exist is :fail' do
+        expected_key = '29d5f92e9ee44d4854d6dfaeefc3dc27d779fdf3/existing-key'
+        stub_s3_response_for_file(expected_key)
+
+        with_tmp_file(named: 'existing-key') do |file_path|
+          expect do
+            run_described_fastlane_action(
+              bucket: test_bucket,
+              key: 'existing-key',
+              file: file_path,
+              if_exists: :fail
+            )
+          end.to raise_error(FastlaneCore::Interface::FastlaneError, "File already exists in S3 bucket #{test_bucket} at #{expected_key}")
+        end
+      end
+
+      it 'logs a message without failing when if_exists is :skip' do
+        expected_key = '29d5f92e9ee44d4854d6dfaeefc3dc27d779fdf3/existing-key'
+        stub_s3_response_for_file(expected_key)
+
+        warnings = []
+        allow(FastlaneCore::UI).to receive(:important) { |message| warnings << message }
+
+        with_tmp_file(named: 'existing-key') do |file_path|
+          key = run_described_fastlane_action(
+            bucket: test_bucket,
+            key: 'existing-key',
+            file: file_path,
+            if_exists: :skip
+          )
+
+          expect(warnings).to eq(["File already exists in S3 bucket #{test_bucket} at #{expected_key}. Skipping upload."])
+          expect(key).to eq(expected_key)
+        end
+      end
+
+      it 'upload the file overriding the existing one when if_exists is :replace' do
+        expected_key = '29d5f92e9ee44d4854d6dfaeefc3dc27d779fdf3/existing-key'
+        stub_s3_response_for_file(expected_key)
+
+        warnings = []
+        allow(FastlaneCore::UI).to receive(:important) { |message| warnings << message }
+
+        with_tmp_file(named: 'existing-key') do |file_path|
+          expect(client).to receive(:put_object).with(body: file_instance_of(file_path), bucket: test_bucket, key: expected_key)
+
+          return_value = run_described_fastlane_action(
+            bucket: test_bucket,
+            key: 'existing-key',
+            file: file_path,
+            if_exists: :replace
+          )
+
+          expect(return_value).to eq(expected_key)
+          expect(Fastlane::Actions.lane_context[Fastlane::Actions::SharedValues::S3_UPLOADED_FILE_PATH]).to eq(expected_key)
+        end
+      end
+
+      it 'throws when if_exists is not one of the expected values' do
+        with_tmp_file(named: 'key') do |file_path|
+          expect do
+            run_described_fastlane_action(
+              bucket: test_bucket,
+              key: 'a8c-key1',
+              file: file_path,
+              if_exists: :invalid
```

Comment:
I had those tests for the custom implementation, then removed them once I moved to use `type: Symbol`. My rationale for removing the few examples I had is the one you highlighted:

> Or do you feel like adding those tests would inherently be us testing `ConfigItem#auto_convert` after all instead of our own code? 🤔

To be fair, I could argue the case for adding a couple of tests: Additional confidence for relatively cheap overhead. But, I wonder whether it would set the precedent for spending more and more time testing Fastlane code that we rely upon. 

I think I'd be more keen to add them if this was an action that we expect to use standalone via `bundle exec fastlane run upload_to_s3` but I don't think that's the case. All the usages I've seen have been within lanes.